client/landing/stepper/declarative-flow/internals/steps-repository/link-in-bio-setup/index.tsx

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+
+import { Gridicon, Button } from '@automattic/components';
+import { useI18n } from '@wordpress/react-i18n';
+import FormattedHeader from 'calypso/components/formatted-header';
+import FormInput from './form-input';
+import type { Step } from '../../types';
+
+import './styles.scss';
+
+const LinkInBioSetup: Step = function LinkInBioSetup( { navigation } ) {
+	const { goNext } = navigation;
+	const siteNameErrorMsg = 'Your site needs a name so your subscribers can identify you.';
+	const siteBriefDescriptionErrorMsg =
+		'Your site needs a brief description so your subscribers can identify you.';
+	const { __ } = useI18n();
+
+	const onChangeSiteName = () => {
+		return;
+	};
+
+	const onChangeSiteDescription = () => {
+		return;
+	};
+
+	return (
+		<div className="step-container">
+			<form>
+				<FormattedHeader align={ 'center' } headerText={ __( 'Set up Link in Bio' ) } />
+				<div className="link-in-bio-setup__form">
+					{ /* replace this component, align with newsletter flow */ }
+					<div className="link-in-bio-setup-form-icon__container">
+						<div className="link-in-bio-setup-form__icon">
+							<Gridicon key={ 'blue' } icon="share-ios" size={ 18 } />
+						</div>
+						<label htmlFor="">{ __( 'Upload site icon' ) }</label>{ ' ' }
+					</div>
+					{ /*  */ }
+					<FormInput
+						inputValue={ '' } // get value from store?
+						isValid={ true }
+						onChange={ onChangeSiteName }
+						errorMessage={ siteNameErrorMsg }
+					/>
+					<FormInput
+						inputValue={ '' } // get value from store?
+						isValid={ true }
+						onChange={ onChangeSiteDescription }
+						errorMessage={ siteBriefDescriptionErrorMsg }
+					/>
+					<div className="link-in-bio-setup-form__field">
+						<label className="link-in-bio-setup-form__label" htmlFor="linkInBioSiteName">
+							{ __( 'Publication address' ) }
+						</label>
+						<div className="link-in-bio-setup-form-field__container">
+							<div className="link-in-bio-setup-form-container__address">
+								{ 'www.test.link ' }
+								<button className="link-in-bio-setup-form__button">{ __( 'Change' ) }</button>
+							</div>
+						</div>
+					</div>
+				</div>
+				<Button className="link-in-bio-setup-form__submit" primary onClick={ goNext }>
```

Comment:
This is usually a job for a `handleSubmit` function that does what it needs to do (probably nothing now, as I think you're waiting for APIs) and then calls the `submit` function from `navigation` component.
For reference you could once again take a look at the `site-options` step [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/site-options/index.tsx#L103) 😄 
`goNext` would usually be used when you skip a step for example.

Reply:
It's great that you already have in mind the `submit` part! You can iterate over this again later, or you could add some skeleton infrastructure now ( an `handleSubmit` function that calls `submit` for this step maybe?)! Your choice, whatever is fine 💪